src/wp-includes/functions.wp-scripts.php

```diff
@@ -224,13 +224,12 @@ function wp_register_script( $handle, $src, $deps = array(), $ver = false, $args
  * @return bool True if the script was successfully localized, false otherwise.
  */
 function wp_localize_script( $handle, $object_name, $l10n ) {
-	global $wp_scripts;
+	$wp_scripts = wp_scripts();
```

Comment:
Note that, if we do this, then the `_wp_scripts_maybe_doing_it_wrong` call below is pointless.

Also, we'd need to remove the `@global` phpdoc above because it's no longer relevant.

But let's discuss the right approach on the ticket first.